ucsb-cs156-s24/proj-happycows-s24-4pm-5
# Github Commit Info Accessible Through /api/systemInfo API Call

## Overview
In this pr we allow a user to make an api call to `/api/systemInfo` to get information about the state of the currently deployment. The info it provides is the truth value of the `H2_CONSOLE_ENABLED`, `SHOW_SWAGGER_UI_LINK`, and value of `SOURCE_REPO` env variables defined in the dokku app, `startQtrYYYYQ`, and `endQtrYYYYQ` which are hardcoded, `commitMessage` and `commitId` which is acquired from a github maven plugin, and `githubUrl` which is created from `SOURCE_REPO` and `commitId`.

## Screenshots 
These screenshots are from my dokku app happycows-garvinyoung-dev as of writing this description, the latest commit is likely to have changed since then.

As you can see, the response body has the correct `commitID`, `commitMessage`, and truth values for `showSwaggerUILink` and `springH2ConsoleEnabled`
![image](https://github.com/ucsb-cs156-s24/proj-happycows-s24-4pm-5/assets/97640757/ae6bf2cc-6640-4cc1-b1ab-fbb55a502d8e)



## Feedback Request
- I just put the startQtr and endQtr as 20243 and 20244 to represent Spring of 2024, are they supposed to be something else or is this fine?

## Validation
Go to: https://happycows-garvinyoung-dev.dokku-05.cs.ucsb.edu/
1.   navigate to the deployed dokku link, you can either access the api call through enabling the swagger link or adding `/api/systemInfo` to the end of the main page url
2. verify that the truth values of the Swagger and H2 Console links are correct as well as the commitMessage and commitId being updated to the lastest commit to this branch

## Tests
In `SystemInfoServiceImplTest.java`, I expanded `test_getSystemInfo()` to cover the additional fields added to the `SystemInfo` class and added `test_githubUrl()` to test that `githubUrl` is correctly created based on `commitId` and `sourceRepo`.

## Linked Issues
Closes #11
